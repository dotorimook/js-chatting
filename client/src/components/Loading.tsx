import React, { FC } from 'react';
import IComponentProps from 'interfaces/IComponentProps';
import styled, { keyframes } from 'styled-components';
import Frame, { FrameCenter } from './Frame';
import img0 from 'assets/img/loading/0.svg';
import img1 from 'assets/img/loading/1.svg';
import img2 from 'assets/img/loading/2.svg';
import img3 from 'assets/img/loading/3.svg';
import theme from 'style/theme';

interface ILoadingProps extends IComponentProps{
  visible: boolean;
};

const loading = keyframes`
  0% {transform: scale(.95, .95);}
  65% {transform: scale(1.05,1.05);}
  100% {transform: scale(.95, .95);}
`

const Dimmer = styled(Frame)`
  position:fixed;
  top:0;
  left:0;
  background:${theme.dimmer};
`;
const LoadingFrame = styled(FrameCenter)`
  position:fixed;
  background:none;
  transition: visibility opacity .1s ease;
  visibility: hidden;
  opacity: 0;
  z-index:99999;
  &.visible {
    visibility: visible;
    opacity:1;
  }
  .title-img {
    animation: ${loading} 1.5s ease-in-out infinite;
  }
  .msg {
    z-index:99999;
  }
`

const Loading:FC<ILoadingProps> = (props:ILoadingProps) => {
  return (
    <LoadingFrame className={`msg-frame ${props.visible? 'visible': ''}`}>
      <Dimmer/>
      <img className='title-img loading' src={[img0,img1,img2,img3][Math.round(Math.random()*3)]} alt='logo'/>
      <div className='msg'>
        로딩중입니다...
      </div>
    </LoadingFrame>
  )
}
export default Loading;