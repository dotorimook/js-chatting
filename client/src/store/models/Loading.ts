import {observable, action} from 'mobx';

export default class Loading {
  @observable isLoading: boolean;
  
  constructor() {
    this.isLoading = false;
  }

  @action
  toggleLoading = () => {
    this.isLoading = !this.isLoading;
  }
}