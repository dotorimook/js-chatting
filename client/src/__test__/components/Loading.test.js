import React from 'react';
import 'setupTest';
import Loading from 'components/Loading';
import {shallow} from 'enzyme';

describe('Loading Test', () => {
  let uiVisible = null;
  let uiInvisible = null;
  global.Math.random = jest.fn(()=>0);

  beforeAll(()=>{
    uiVisible = shallow(<Loading visible={true}/>);
    uiInvisible = shallow(<Loading visible={false}/>);

  });

  it('properly renders', () => {
    expect(uiVisible.find('.msg-frame.visible').length).toBeGreaterThan(0);
    expect(uiInvisible.find('.msg-frame.visible').length).toBe(0)
  });

  it('matches snapshot', () => {
    expect(uiVisible).toMatchSnapshot();
  });

});