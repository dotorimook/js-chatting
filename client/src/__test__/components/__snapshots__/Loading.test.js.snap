// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Test matches snapshot 1`] = `
<Styled(Styled(styled.div))
  className="msg-frame visible"
>
  <Styled(styled.div) />
  <img
    alt="logo"
    className="title-img loading"
    src="0.svg"
  />
  <div
    className="msg"
  >
    로딩중입니다...
  </div>
</Styled(Styled(styled.div))>
`;
